src/main/java/org/truffleruby/core/regexp/ClassicRegexp.java

```diff
@@ -104,6 +104,11 @@ public void warn(String s) {
     }
 
     static Regex getRegexpFromCache(RubyContext runtime, RopeBuilder bytes, Encoding enc, RegexpOptions options) {
+        if (runtime == null) {
```

Comment:
Not for this PR:
I wish we had a clearer condition to express "parsing during native image building".
Maybe a static variable or simply a comment next to those checks.

Reply:
Maybe parsing should just be totally context independent in the first place? Structures created the parser should be lower level and then used by context dependent structures; like `Rope` is.

Reply:
That would probably be ideal.
But it seems not easy.

Reply:
We have `ParserRopeOperations.isBuildingNativeImage`, which does what @eregon is describing. Thus far it's only been needed for `ParserRopeOperations`, so it's private. But feel free to move it around.